### ISSUE TITLE
incompatible with react 18

### ISSUE THREAD

**Issue by md-redwan-hossain** [tawk/tawk-messenger-react#10]
I tried to install tawk react package but it throws a error saying incompatible with react dom 18. See the atached error log and screenshot.
https://termbin.com/gb1b
![react_bug](https://user-images.githubusercontent.com/15859460/167264424-2f4eab9f-f6de-448c-8c23-676dc6d852d5.png)

**Comment by jaoaustero** [comment]
Hi @redwan-hossain we are aware on the error and I already submit a pull request on the team, this will be on under review and test. Pull request #11 

Thank you

**Comment by jaoaustero** [comment]
Hi @redwan-hossain We already now supporting version 18 of react, we have new updates on the plugin
that will resolve the issue

https://www.npmjs.com/package/@tawk.to/tawk-messenger-react

**Comment by soheb-mpc91** [comment]
**### I AM USING TYPESCRIPT.**

@jaoaustero - I am trying to use tawk in react@18.1.0, Installation is success but i am getting type error when i try to import
with code `import TawkMessengerReact from '@tawk.to/tawk-messenger-react';`

<img width="1042" alt="image" src="https://user-images.githubusercontent.com/105640177/171131874-f041fc5f-0d4d-453c-9675-c8ecf23ea857.png">


### I am also trying to install with npm package `npm i --save-dev @types/tawk.to__tawk-messenger-react` but getting below error:
<img width="801" alt="image" src="https://user-images.githubusercontent.com/105640177/171132883-4cf5ebdb-d2b1-4874-b55f-5bb8710ffa18.png">

**Comment by jaoaustero** [comment]
Hi @soheb-mpc91 currently we don't yet support typescript but this was the next thing we will update

**Comment by soheb-mpc91** [comment]
@jaoaustero - Any update on https://github.com/tawk/tawk-messenger-react/issues/10#issuecomment-1141854908 ?

**Comment by jleknes** [comment]
Have you considered supporting react 17 in the latest version of the plugin? Now you have set a 18.1.0 as the minimum version of `react` and `react-dom` in `peerDependencies`. This throws a warning if you try to use the plugin with react 17. And do you really need the peer dependency to `react-scripts`?

**Comment by jaoaustero** [comment]
@jleknes You can still use this plugin in react 17 by using the lower version 1.0.0

**Comment by jleknes** [comment]
@jaoaustero: Yes, but version 1.0.0 has lots of outdated dependencies with critical vulnerabilities.

Is version 2.0.1 really incompatible with react 17 or have you just set the peerDependency too aggressively?

**Comment by smith558** [comment]
> Hi @soheb-mpc91 currently we don't yet support typescript but this was the next thing we will update

Just discovered this. This is quite unfortunate for us.
